osu.Game/Overlays/OverlayRulesetTabItem.cs

```diff
@@ -64,29 +64,33 @@ public OverlayRulesetTabItem(RulesetInfo value)
         [BackgroundDependencyLoader]
         private void load()
         {
-            updateState();
+            UpdateState();
         }
 
         protected override bool OnHover(HoverEvent e)
         {
             base.OnHover(e);
-            updateState();
+            UpdateState();
             return true;
         }
 
         protected override void OnHoverLost(HoverLostEvent e)
         {
             base.OnHoverLost(e);
-            updateState();
+            UpdateState();
         }
 
-        protected override void OnActivated() => updateState();
+        protected override void OnActivated() => UpdateState();
 
-        protected override void OnDeactivated() => updateState();
+        protected override void OnDeactivated() => UpdateState();
 
-        private void updateState()
+        protected void UpdateState()
         {
             text.Font = text.Font.With(weight: Active.Value ? FontWeight.Bold : FontWeight.Medium);
+
+            if (!Enabled.Value)
+                return;
```

Comment:
I don't understand what's going on with `UpdateState()` - why is it exposed as protected and why does it have this guard in the middle?

Reply:
Item deselecting happens after disabling, so we don't want to update the colour here. Tricky stuff, but my best thoughts so far, will revisit it

Reply:
Okay, should be fixed